### DISCUSSION TITLE
Running a SIP Proxy

### DISCUSSION THREAD

**Discussion by github4tiri** [fonoster/yaps#487] [General]
Due to firewall restrictions, it is forbidden to run outgoing connections from SIP Client Devices directly to the SIP Provider.
Is there a way using routr as a SIP Proxy (like a Squid for Web-Connections), and how could an example configuration look like?
Thanks for your suggestions and ideas in advance!

**Comment by psanders** [comment]
@github4tiri I am not familiar with Squid, but if you have firewall restrictions those will also affect Routr. What sort of firewall restrictions are you facing? Is it for a home network?

**Comment by psanders** [comment]
I will close this conversation to maintain organization. However, if you need further assistance in the future, feel free to re-open it.

**Comment by github4tiri** [comment]
A proxy makes it possible to route all traffic through a specific application, so as squid does it for webtraffic, I supposed using fonostr as a sip proxy just only to configure the target of my client being the fonostr server, and the fonostr will route the trffic wherever needed. So no sip client in my local network needs a rule to connect via SIP to an address outside the home network, and the fonostr sip proxy will be able to connect to all.
Do you have any idea how to do that with SIP?

Btw. here is https://en.wikipedia.org/wiki/Proxy_server

**Reply by psanders** [reply]
Yes, Routr can be used for that purpose. You can set up Routr to direct your Softphone/IPPhone to a SIP Service Provider, such as Twilio, VoIP.ms, Bandwidth, etc. Additionally, Routr can be used for making internal calls within your network. To get started, you may refer to the following link: https://routr.io/docs/guides/basic-setup/
